WcaOnRails/app/models/competition.rb

```diff
@@ -266,11 +294,37 @@ def registration_must_close_after_it_opens
     end
   end
 
+  def use_wca_as_website=(value)
+    @use_wca_as_website = ActiveRecord::Type::Boolean.new.type_cast_from_database(value)
+  end
+
+  def use_wca_as_website
+    @use_wca_as_website || website == url
+  end
+
+  before_validation :assign_wca_website_url
+  private def assign_wca_website_url
+    if use_wca_as_website && website != url
+      self.website = self.url
+    end
+  end
+
+  after_save :update_website
+  private def update_website
+    if use_wca_as_website && website != url
+      update_column :website, url
```

Comment:
About line 298:
it assigns `@use_wca_as_website` using `ActiveRecord::Type::Boolean.new.type_cast_from_database` that converts values like `"0"` and `"1"`. Such values are sent via form by checkbox and then passed by controller to be assigned.

Your code makes sense and it's actually what we do above (`assign_wca_website_url`), but if `competition.use_external_website` is `true` then `competition.website is already assigned`, so I think an if statement is ok. In `assign_wca_website_url` I've added the second condition to not re-assign the same thing before each validation if wca website is used.

The `update_website` is necessary to update website if *the wca one* changes, so when `id` changes.